packages/react-client/src/ReactFlightReplyClient.js

```diff
@@ -1279,16 +1279,16 @@ export function createBoundServerReference<A: Iterable<any>, T>(
     const location = metaData.location;
     if (location) {
       const functionName = metaData.name || '';
-      const [, filename, line, col] = location;
+      const [, filename, , , enclosingLine, enclosingCol] = location;
       const env = metaData.env || 'Server';
       const sourceMap =
         findSourceMapURL == null ? null : findSourceMapURL(filename, env);
       action = createFakeServerFunction(
         functionName,
         filename,
         sourceMap,
-        line,
-        col,
+        enclosingLine,
+        enclosingCol,
         env,
         action,
```

Comment:
We want the fake function to be generated based on both like we do in `ReactFlightClient`. 

Right now this causes devtools to point to the beginning of the module since that's the enclosing line number of the `registerServerReference` call.

https://github.com/user-attachments/assets/2c9e29be-963c-49fe-b193-8ff7b5a775a3


Before this change we pointed at the declaration of `increment` since that's where `registerServerReference` call sourcemaps to:

```
1   | 'use server'
      ^ maps to 'use server'
      enclosing line number
      we don't want that
... |
29  | registerServerReference(increment, "...", "increment");
      ^ maps maps to `function increment() {}`
      line number
      we want that

```

Reply:
I guess we never want the enclosing because the line in the `location` field always represents the enclosing, which is the only thing we encode here.

Reply:
Reverted that part.

Reply:
There's never anything that executes inside the function so there's never anything that gets the name from source mapping. But if for example printing it in the console printed the name from the source mapped source then it would source map it but even then it would be correct because it is the enclosing line that we're modeling in the fake function. The registerServerReference is at the enclosing location and its callsite location is therefore modeling the enclosing location which we are then modeling with the fake function's enclosing location.

We just have to be careful if we do something like extract the enclosing and passing that as "location" meta data. I think maybe that's the modeling issue here. The server should actually encode the location as the enclosing line.